src/libcollections/vec_deque.rs

```diff
@@ -175,9 +175,11 @@ impl<T> VecDeque<T> {
     /// (abs(dst - src) + len) must be no larger than cap() (There must be at
     /// most one continuous overlapping region between src and dest).
     unsafe fn wrap_copy(&self, dst: usize, src: usize, len: usize) {
-        debug_assert!(
-            (if src <= dst { dst - src } else { src - dst }) + len <= self.cap(),
-            "dst={} src={} len={} cap={}", dst, src, len, self.cap());
+        #[allow(dead_code)]
+        fn diff(a: usize, b: usize) -> usize {if a <= b {b - a} else {a - b}}
```

Comment:
not sure its worth very much making this function instead of keeping it inlined.